From b5d74989c8278c20d463701f8f39804596579dd8 Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Tue, 12 Sep 2017 08:22:40 -0400
Subject: [PATCH 80/82] crypto: keystone: enable EVICT flag by default

Enable the EVICT flag by default to allow the key to be changed for
an AEAD (crypto_aead_setkey called more than once), like what is done
during crypto selftests.

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
---
 drivers/crypto/keystone-sa-utils.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/crypto/keystone-sa-utils.c b/drivers/crypto/keystone-sa-utils.c
index c3a2275..03eed2d 100644
--- a/drivers/crypto/keystone-sa-utils.c
+++ b/drivers/crypto/keystone-sa-utils.c
@@ -540,15 +540,12 @@ int sa_init_sc(struct sa_ctx_info *ctx, const u8 *enc_key,
 	hash_size = roundup(hash_size, 8);
 
 	sa_set_swinfo(first_engine, ctx->sc_id, ctx->sc_phys, 1, 0,
-#ifndef TEST
-		      0,
-#else
 	/*
 	 * For run-time self tests in the cryptographic
-	 * algorithm manager framework
+	 * algorithm manager framework the EVICT flag is required.
+	 * EVICT is also required if the key gets changed for the context.
 	 */
 		      SA_SW_INFO_FLAG_EVICT,
-#endif
 		      ctx->rx_compl_qid, ctx->rx_flow, hash_size, swinfo);
 
 	sa_dump_sc(sc_buf, ctx->sc_phys);
-- 
1.9.1

